lib/compress.js

```diff
@@ -244,7 +244,11 @@ merge(Compressor.prototype, {
                 }
                 if (node instanceof AST_ForIn) {
                     if (node.init instanceof AST_SymbolRef) {
-                        node.init.definition().fixed = false;
+                        var d = node.init.definition();
+                        d.references.push(node.init);
+                        d.fixed = false;
```

Comment:
It's because with `for (a in obj) {...}`, `node.init` is just `AST_SymbolRef` which is `a`.

In theory I could extend `isModified()` to cover for-in loops, but putting it here is easier given the structure of the code.